File: cases/taint/go/context-pair-depth2-positive/context_pair_depth2_positive.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: context-pair-depth2-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: context-pair-depth2-sink

func helper(value string) string { // DFB-WITNESS: context-pair-depth2-helper
	return value
}

func wrapper(value string) string {
	return helper(value)
}

func outerTainted() string {
	return wrapper(dfb_source())
}

func outerClean() string {
	return wrapper("clean")
}

func run() {
	tainted := outerTainted()
	clean := outerClean()
	_ = clean
	dfb_sink(tainted)
}

File: cases/taint/rust/context-pair-depth2-positive/context_pair_depth2_positive.rs
fn dfb_source() -> i32 { // DFB-SOURCE: context-pair-depth2-input
    1
}

fn dfb_sink(value: i32) {} // DFB-SINK: context-pair-depth2-sink

fn helper(value: i32) -> i32 { // DFB-WITNESS: context-pair-depth2-helper
    value
}

fn wrapper(value: i32) -> i32 {
    helper(value)
}

fn outer_tainted() -> i32 {
    wrapper(dfb_source())
}

fn outer_clean() -> i32 {
    wrapper(0)
}

fn run() {
    let tainted = outer_tainted();
    let clean = outer_clean();
    dfb_sink(tainted);
}
